### DISCUSSION TITLE
Add ability to change the title of any panel

### DISCUSSION THREAD

**Discussion by iliabaranov** [foxglove/community#459] [Q&A]
Hello Foxglove team!

I'd love to be able to change the title of panels. In particular, I was thinking of the new gauges, as I want to have a few different ones, and it's hard to tell between them without labels. But the ability to change titles of other panels would also be very useful.

Thanks!

**Comment by jessicaaustin** [comment]
+1 for this, Indicator has a similar problem. Would love to replace `Gauge` or `Indicator` with a title instead, like we can for plots

![Selection_999(1253)](https://user-images.githubusercontent.com/1159930/189431210-830a2e11-3513-4545-9dba-3855c91f33bf.png)

**Comment by jtbandes** [comment]
@foxymiles @2metres can you provide some input on how we might implement this? Would the title be editable inline? Would this be part of the settings API or some separate API? Would the panel be able to change its own title or can only the user do it?

**Comment by foxymiles** [comment]
> @foxymiles @2metres can you provide some input on how we might implement this?

I took a stab at an approach that bypasses the actual panel implementation entirely but it was not favorably received.

https://github.com/foxglove/studio/pull/3937

Is it necessary for the panel implementation to be aware of its title or can we implement in a general way that doesn't concern the panel internals?

**Comment by jtbandes** [comment]
> Is it necessary for the panel implementation to be aware of its title or can we implement in a general way that doesn't concern the panel internals?

It seems more useful for a panel to be able to _set_ its own title than to read its own title. But neither of these seem very important and we could probably skip them for a first version of this feature.

**Comment by sharst** [comment]
My usecase would be the following: I have a lot of indicators on the same screen which display values from deeply nested messages. Therefore the message path is not really descriptive anymore. It would be extremely helpful to be able to name these panels somehow - and in a first version I wouldn't really care if the panel is aware of it's name or not. 
A similar situation happens when you use a raw message panel where the message name is not self-explanatory, or when you again filter for only some values of a nested message.

I'm a bit surprised that the issue on github got closed and transferred to a discussion although there seems to be legitimate interest - what's the policy on that?
